rust/bundle-helper/src/main.rs

```diff
@@ -0,0 +1,187 @@
+use futures::StreamExt;
+use snafu::{OptionExt, ResultExt, Snafu};
+use stackable_operator::client;
+use stackable_operator::client::Client;
+use stackable_operator::error;
+use stackable_operator::error::OperatorResult;
+use stackable_operator::k8s_openapi::api::core::v1::ConfigMap;
+use stackable_operator::kube::api::ListParams;
+use stackable_operator::kube::runtime::controller::Context;
+use stackable_operator::kube::runtime::controller::ReconcilerAction;
+use stackable_operator::kube::runtime::Controller;
+use stackable_operator::kube::Api;
+use stackable_operator::logging::controller::report_controller_reconciled;
+use stackable_operator::logging::controller::ReconcilerError;
+use std::fs::create_dir_all;
+use std::fs::rename;
+use std::fs::File;
+use std::io::prelude::*;
+use std::path::Path;
+use std::sync::Arc;
+use std::time::Duration;
+use std::env;
+use strum::{EnumDiscriminants, IntoStaticStr};
+
+#[derive(Snafu, Debug, EnumDiscriminants)]
+#[strum_discriminants(derive(IntoStaticStr))]
+#[allow(clippy::enum_variant_names)]
+pub enum Error {
+    #[snafu(display("object defines no version"))]
+    ObjectHasNoVersion,
+    #[snafu(display("opa bundle has no name"))]
+    OpaBundleHasNoName,
+    #[snafu(display("opa bundle dir error"))]
+    OpaBundleDir { source: std::io::Error },
+    #[snafu(display("missing namespace to watch"))]
+    MissingWatchNamespace,
+}
+
+impl ReconcilerError for Error {
+    fn category(&self) -> &'static str {
+        ErrorDiscriminants::from(self).into()
+    }
+}
+pub struct Ctx {
+    pub root: String,
+    pub incomming: String,
+}
+
+const WATCH_NAMESPACE_ENV: &str = "WATCH_NAMESPACE";
+
+#[tokio::main]
+async fn main() -> Result<(), error::Error> {
+    stackable_operator::logging::initialize_logging("OPA_BUNDLE_HELPER_LOG");
+
+    let client = client::create_client(Some("opa.stackable.tech".to_string())).await?;
+
+    match env::var(WATCH_NAMESPACE_ENV) {
+        Ok(namespace) => {
+            create_controller(client, namespace, "/bundles/active", "/bundles/incomming").await?;
+        }
+        Err(_) => {
+            tracing::error!(
+                "Missing namespace to watch. Env var [{}] is probably not defined.",
+                WATCH_NAMESPACE_ENV
+            );
+        }
+    }
+
+    Ok(())
+}
+
+async fn create_controller(
+    client: Client,
+    namespace: impl Into<String>,
+    root: impl Into<String>,
+    incomming: impl Into<String>,
+) -> OperatorResult<()> {
+    let configmaps_api: Api<ConfigMap> = client.get_namespaced_api(namespace.into().as_ref());
+
+    let controller = Controller::new(
+        configmaps_api,
+        ListParams::default().labels("opa.stackable.tech/bundle=true"),
```

Comment:
Do we expect to support different types of bundles or anything like that? Having a label with value true seems a bit ... weird. I'd rather check if the label `opa.stackable.tech/bundle` is present and then match on the content..